Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/ejb/impl/AuditEJBLoggerImpl.java

```diff
@@ -0,0 +1,105 @@
+ *       copyright notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the documentation
+ *       and/or other materials provided with the distribution.
+ *     * Neither the name of the United States Government nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.audit.ejb.impl;
+
+import gov.hhs.fha.nhinc.audit.ejb.AuditEJBLogger;
+import gov.hhs.fha.nhinc.audit.transform.AuditTransforms;
+import gov.hhs.fha.nhinc.auditrepository.nhinc.proxy.AuditRepositoryProxyObjectFactory;
+import gov.hhs.fha.nhinc.common.auditlog.LogEventRequestType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
+import java.util.Properties;
+import javax.ejb.Asynchronous;
+import javax.ejb.Stateless;
+import org.apache.log4j.Logger;
+
+/**
+ *
+ * @author achidamb
+ * @param <T> Request
+ * @param <K> Response
+ */
+@Stateless
+public class AuditEJBLoggerImpl<T, K> implements AuditEJBLogger<T, K> {
+
+    private static final Logger LOG = Logger.getLogger(AuditEJBLoggerImpl.class);
+
+    /**
+     * EJB Asynchronous call to handle AuditRepository client
+     *
+     * @param request Request to be audited
+     * @param assertion assertion to be audited
+     * @param target target community
+     * @param direction defines the Outbound/Inbound message
+     * @param _interface Entity, Adapter or Nwhin
+     * @param isRequesting true for initiator, false for responder
+     * @param webContextProperties Properties loaded from message context
+     * @param serviceName Name of the Service being audited
+     */
+    @Asynchronous
+    @Override
+    public void auditRequestMessage(T request, AssertionType assertion, NhinTargetSystemType target, String direction, String _interface, Boolean isRequesting, Properties webContextProperties, String serviceName, AuditTransforms transforms) {
```

Comment:
Please fix any code you've touched where the line length exceeds 120 characters.